README.md

```diff
@@ -1,148 +1,213 @@
-running on, you need to tell Crate to publish the address of the docker host for discovery.
+Crate recognizes a few environment variables like `CRATE_HEAP_SIZE`
+that need to be set with the `--env` option before the actual Crate
+core starts. You may want to [assign about half of your memory
+it](https://crate.io/docs/reference/en/latest/configuration.html#crate-heap-size)
+as a rule of thumb to Crate like this:
 
-    docker run -d -p 4200:4200 -p 4300:4300 crate/crate crate -Des.network.publish_host=host1.example.com:
+```console
+# docker run -d --env CRATE_HEAP_SIZE=32g crate/crate
+```
 
-If you change the transport port from the default ``4300`` to something else, you also need to pass
-the publish port to Crate.
+## Open Files
 
-    docker run -d -p 4200:4200 -p 4321:4300 crate/crate crate -Des.transport.publish_port=4321
+Depending on the size of your installation Crate opens a lot of
+files. You can check the number of open files with `ulimit -n`. It
+depends on your host operation system. To increase the number start
+containers with the option `--ulimit nofile=65535:65535`:
 
-### Example Usage in a Multinode Setup
+## Multicast
 
-    HOSTS='crate1.example.com:4300,crate2.example.com:4300,crate3.example.com:4300'
-    HOST=crate1.example.com
-    docker run -d \
-        -p 4200:4200 \
-        -p 4300:4300 \
-        --name node1 \
-        --volume /mnt/data:/data \
-        --env CRATE_HEAP_SIZE=8g \
-        crate/crate:latest \
-        crate -Des.cluster.name=cratecluster \
+Crate uses multicast for node discovery by default. This means nodes
+started in the same multicast zone will discover each other
+automatically. However, Docker multicast support between containers on
+different hosts depends on the overlay network driver. If that does
+not support multicast, you have to [enable unicast in your custom
+`crate.yml`](https://crate.io/docs/reference/best_practice/multi_node_setup.html).
+
+Crate publishes the hostname it runs on for discovery within the
+cluster. If the address of the docker container differs from the
+actual host the docker image is running on -- this is the case if you
+do port mapping to the host via the `-p` option, you need to tell
+Crate to publish the address of the docker host instead:
+
+```console
+# docker run -d -p 4200:4200 -p 4300:4300 crate/crate \
+    crate -Des.network.publish_host=host1.example.com
+```
+
+If you change the transport port from the default `4300` to something
+else, you also need to pass the publish port to Crate by adding
+`-Des.transport.publish_port=4321` to your command.
+
+## Example Usage in a Multihost Setup
+
+To start a Crate cluster in containers distributed to three hosts
+without multicast enabled, run this command on the first node and
+adapt container and node names on the two other nodes:
+
+```console
+# HOSTS="crate1.example.com:4300,crate2.example.com:4300,crate3.example.com:4300"
+# HOST="crate1.example.com"
+# docker run -d -P \
```

Comment:
this will not work like this. -p 4200:4200 -p 4300:4300 is required since Crate expects other nodes to have port 4300 open for communication